### PR TITLE
Fix the preview link purge job

### PR DESCRIPTION
**Pull request by aschempp**
Fixes https://github.com/contao/contao/issues/5049

### CODE REVIEW

**Review by Toflar — DISMISSED** [review]

**Comment by ausi** [comment]
Should we maybe check for both, `createdAt` **and** `expiresAt` to make sure we don’t break customized usages like the one outlined here: https://github.com/contao/contao/issues/5089#issuecomment-1206149076

**Comment by aschempp** [comment]
it is intentionally not possible, but you can always bypass the cronjob if you wanna break the system 🤷

**Comment by ausi** [comment]
I don’t think that people who adjust the DCA would know that they have to adapt the cron job too. What would be the downside of checking `expiresAt` too (and only deleting when both are out of time)

File: core-bundle/src/Cron/PurgePreviewLinksCron.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ public function __construct(Connection $connection)
     public function __invoke(): void
     {
         $this->connection->executeStatement(
-            'DELETE FROM tl_preview_link WHERE createdAt<=UNIX_TIMESTAMP(DATE_ADD(NOW(), INTERVAL 31 DAY))'
+            'DELETE FROM tl_preview_link WHERE createdAt<=UNIX_TIMESTAMP(DATE_SUB(NOW(), INTERVAL 31 DAY))'
```
Comment (ausi) [inline core-bundle/src/Cron/PurgePreviewLinksCron.php:38]:
```suggestion
            'DELETE FROM tl_preview_link WHERE createdAt<=UNIX_TIMESTAMP(DATE_SUB(NOW(), INTERVAL 31 DAY)) AND expiresAt<=UNIX_TIMESTAMP(NOW())'
```
_(untested)_

Comment (aschempp) [reply core-bundle/src/Cron/PurgePreviewLinksCron.php:38]: This change would make it easy to create a preview link that lives indefinitely, and I don't think we should ever allow for that for security reasons.

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/Cron/PurgePreviewLinksCron.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Cron/PurgePreviewLinksCron.php:38]: I have created a separate ticket hereto (see #5184).

**Review by leofeyer — APPROVED** [review]

**Pull request merged**